src/history/hs.c

```diff
@@ -799,6 +815,8 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
             /* If we've seen a smaller timestamp before, use that instead. */
             if (min_insert_ts < upd->start_ts) {
                 /*
+                 * Only fix out of order timestamp if we haven't seen an update without timestamp.
```

Comment:
The comment doesn't seem accurate. We still fix them up, we just don't reset them back to WT_TS_NONE (that's what it looks like the code is doing anyway, and what I expect it to do).

Reply:
Fixed. @agorrod Can I merge it?